docs/src/index.md

```diff
@@ -4,6 +4,147 @@
 Depth = 4
 ```
 
+## Getting Started
+
+> Yet another JSON package for Julia; this one is for speed and slick struct mapping
+
+JSON3 provides two main functions: [`JSON3.read`](@ref) and [`JSON3.write`](@ref).  These allow in the basic case, reading a JSON string into a [`JSON3.Object`](@ref) or [`JSON3.Array`](@ref), which allow for dot or bracket indexing and can be copied into base `Dicts` or `Vectors` if needed.  The slick struct mapping allows reading a JSON string directly into (almost) any type you wish and then writing directly from those types into JSON as well.
+
+### Examples
+
+#### Basic reading and writing
+```@example
+using JSON3 # hide
+json_string = """{"a": 1, "b": "hello, world"}"""
+
+hello_world = JSON3.read(json_string)
+
+# can access the fields with dot or bracket notation
+println(hello_world.b)
+println(hello_world["a"])
+
+JSON3.write(hello_world)
+```
+
+#### Write with pretty printing
+```@example
+using JSON3 # hide
+json_string = """{"a": 1, "b": "hello, world"}"""
+
+hello_world = JSON3.read(json_string)
+JSON3.pretty(JSON3.write(hello_world))
+```
+
+#### Read and write from/to a file
+```jl
+json_string = read("my_file.json", String)
+
+hello_world = JSON3.read(json_string)
+
+open("my_new_file.json", "w") do io
+    JSON3.write(io, hello_world)
+end
+```
+
+#### Read JSON into a type
+
+See more details on the types that are provided and how to customize parsing [below](#Builtin-types).
```

Comment:
I think this should link to `## Struct API` instead?

Reply:
I linked it to the builtin-types as that serves as in intro for what will happen automagically vs when struct mapping is needed, but I agree that given this example context most will likely just want to jump straight to `## Struct API`.